### PR TITLE
Add plumbing for pr param in UI

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Extension of #785 that includes the UI plumbing needed.

## Review Information
Only need to review the last 2 commits.

Find a PR that has loaded results into wpt.fyi, and load ?pr=[PR number]
e.g. [/results/?pr=14143](https://prs-ui-dot-wptdashboard-staging.appspot.com/results/?pr=14143)

### CODE REVIEW

File: webapp/test_results_handler.go [hunk 1]
```diff
@@ -17,13 +17,15 @@ import (
...
 type testRunUIFilter struct {
+	PR       *int // GitHub PR to fetch the results for.
 	Products string
 	Labels   string
 	SHA      string
 	Aligned  bool
 	MaxCount *int
 	From     string
 	To       string
+	Search   string
```
Comment (foolip) [inline webapp/test_results_handler.go]: Drive-by missing field, or accidental edit?

File: webapp/test_results_handler.go [hunk 2]
```diff
@@ -90,36 +82,46 @@ func parseTestResultsUIFilter(r *http.Request) (filter testResultsUIFilter, err
...
-		experimentalAlignedExceptEdge := shared.IsFeatureEnabled(ctx, "experimentalAlignedExceptEdge")
-		masterRunsOnly := shared.IsFeatureEnabled(ctx, "masterRunsOnly")
-		if experimentalByDefault {
-			if experimentalAlignedExceptEdge {
-				testRunFilter = testRunFilter.OrAlignedExperimentalRunsExceptEdge()
+	filter.PR, err = shared.ParsePRParam(r)
+	if err != nil {
+		return filter, err
+	} else if filter.PR != nil {
+		one := 1
```
Comment (foolip) [inline webapp/test_results_handler.go]: Haha, one of the least debatable lines of code I've seen in a while :D

File: webapp/test_results_handler.go [hunk 3]
```diff
@@ -90,36 +82,46 @@ func parseTestResultsUIFilter(r *http.Request) (filter testResultsUIFilter, err
...
 
 	diff, err := shared.ParseBooleanParam(r, "diff")
 	if err != nil {
 		return filter, err
 	}
 	filter.Diff = diff != nil && *diff
-	diffFilter, _, err := shared.ParseDiffFilterParams(r)
-	if err != nil {
-		return filter, err
+	if filter.Diff {
```
Comment (foolip) [inline webapp/test_results_handler.go]: I can't quite tell if anything in this PR made this change necessary, but looks reasonable :)

File: webapp/test_results_handler.go (hunk 1 above)
Comment (lukebjerring) [reply webapp/test_results_handler.go]: Drive-by cleanup.

File: webapp/test_results_handler.go (hunk 2 above)
Comment (lukebjerring) [reply webapp/test_results_handler.go]: I find it extremely frustrating that you can't point to the address of a constant.

File: webapp/test_results_handler.go (hunk 3 above)
Comment (lukebjerring) [reply webapp/test_results_handler.go]: Drive-by cleanup.
